### PR TITLE
fix: Invalid phone error on prefill with plus(+) sign

### LINKED ISSUE #15047: Invalid Phone Error
Invalid phone error when prefillng a phone number as phone=+919999999999. 

+ is being converted to space. Looks like it started happening recently maybe when we started using Nextjs App Router

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
- Fixes #15047 
- Fixes CAL-3708

[Loom showing it is fixed ](https://www.loom.com/share/1b9a0392c9cd48398569ea5cdcc9b6f7)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?
- Make an event-type with a booking question with identifier 'phone' and chose type = Phone
- Prefill that event with query param `phone=+919999999999`
- See the loom

### CODE REVIEW

**Comment by hariombalhara** [comment]
* **#15046** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/15046?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hariombalhara and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/features/bookings/lib/getBookingResponsesSchema.test.ts [hunk 1]
```diff
@@ -506,7 +506,7 @@ describe("getBookingResponsesSchema", () => {
         })
       );
     });
-    test(`should succesfull give responses if phone type field value is valid`, async ({}) => {
+    test(`should successful give responses if phone type field value is valid`, async ({}) => {
```
Comment (hariombalhara) [inline packages/features/bookings/lib/getBookingResponsesSchema.test.ts:509]:
```suggestion
    test(`should successfully give responses if phone type field value is valid`, async ({}) => {
```

File: packages/features/bookings/lib/getBookingResponsesSchema.ts [hunk 2]
```diff
@@ -90,6 +90,10 @@ function preprocess<T extends z.ZodType>({
             parsedValue = JSON.parse(value);
           } catch (e) {}
           newResponses[field.name] = parsedValue;
+        } else if (field.type === "phone") {
+          // + in URL could be replaced with space, so we need to replace it back
+          // Replace the space(s) in the beginning with + as it is supposed to be provided in the beginning only
+          newResponses[field.name] = value.replace(/^ +/, "+");
```
Comment (hariombalhara) [inline packages/features/bookings/lib/getBookingResponsesSchema.ts:96]: + in RegExp is a special character by default. So, it isn't replacing' +'. It is replacing '1 or more spaces'

**Review by Udit-takkar — APPROVED** [review]
LGTM.

**Pull request merged**